platforms/springboot/src/main/java/io/hawt/springboot/HawtioPluginController.java

```diff
@@ -0,0 +1,45 @@
+package io.hawt.springboot;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+import org.springframework.web.util.UriComponents;
+
+import java.util.List;
+
+public class HawtioPluginController {
+
+    private final EndpointPathResolver endpointPath;
+    private List<HawtioPlugin> plugins;
+
+    public HawtioPluginController(final EndpointPathResolver endpointPath) {
+        this.endpointPath = endpointPath;
+    }
+
+    public void setPlugins(final List<HawtioPlugin> plugins) {
+        this.plugins = plugins;
+    }
+
+    public String forwardHawtioRequestToIndexHtml(HttpServletRequest request) {
+        final String path = endpointPath.resolve("hawtio");
+
+        if (request.getRequestURI().equals(path)) {
+            String query = request.getQueryString();
+            if (query != null && !query.isEmpty()) {
+                return "redirect:" + path + "/index.html?" + query;
+            }
+            return "redirect:" + path + "/index.html";
+        }
+
+        final UriComponents uriComponents = ServletUriComponentsBuilder.fromPath(path)
+            .path("/index.html")
+            .build();
+        return "forward:" + uriComponents.getPath();
+    }
+
+//    @RequestMapping("/plugin")
```

Comment:
leftover?

Reply:
and what about `@ResponseBody`?

Reply:
the comment was a leftover, the @ResponseBody is needed, this way Spring Boot serialisation will be used, otherwise there may be issues (I had similar issues when implementing something similar in Camel)